src/main/config.cpp

```diff
@@ -0,0 +1,75 @@
+#include <cstdio>
+#include <iostream>
+#include <sstream>
+#include "gflags/gflags.h"
+#include "config.hpp"
+#include "exec_monitor.hpp"
+
+const char DELIMITER = ',';
+
+/*
+	FLAGS SECTION
+*/
+
+// EXEC FLAG - MONITOR PROCESS EXECUTION
+DEFINE_string(ppid, "", "Specifies the parent pid(s) of the processes to monitor using comma-separated values.");
+DEFINE_string(name, "", "Specifies the name(s) of the processes to monitor using comma-separated values.");
+DEFINE_int32(n, 0, "Specifies the number of processes to monitor. The program will stop after n processes were executed.");
+DEFINE_int32(t, 0, "Specifies the number of seconds to monitor. The program will stop after t seconds.");
+
+// DATA EXPORT FLAGS
+DEFINE_string(save, "", "Specifies the format in which to save the terminal output in a separate file. Available formats are csv and protobuf.");
+DEFINE_validator(save, Config::check_save_and_export_format);
+DEFINE_string(export, "", "Specifies the format in which to export data in a separate file. Using export will not print to stdout. Available formats are csv and protobuf.");
+DEFINE_validator(export, Config::check_save_and_export_format);
+
+/*
+	END OF FLAGS SECTION
+*/
+
+Config::Config() {}
+
+int Config::exec_monitor_parse_args_run()
+{
+	if (!gflags::GetCommandLineFlagInfoOrDie("ppid").is_default)
+	{
+		std::string token;
+		std::istringstream tokenStream(FLAGS_ppid);
+		int ppid;
+		while (std::getline(tokenStream, token, DELIMITER)) {
+			try {
+				ppid = stoi(token);
+			}
+			catch(std::invalid_argument& e) {
+				std::cerr << "Unexpected ppid format: " << token << "\n";
+				return -1;
+			}
+			ppid_list.push_back(ppid);
+		}
+	}
+
+	if (!gflags::GetCommandLineFlagInfoOrDie("name").is_default)
+	{
+		std::string token;
+		std::istringstream tokenStream(FLAGS_name);
+		while (std::getline(tokenStream, token, DELIMITER))
+			name_list.push_back(token);
+	}
+
+	if (!gflags::GetCommandLineFlagInfoOrDie("n").is_default)
+		n_proc = (int)FLAGS_n;
+
+	std::cout << "Monitoring process executions...\n";
+	ExecMonitor exec_monitor;
+	int ret = exec_monitor.run();
```

Comment:
Your Config class is a leaky abstraction. First of all because its caller (main) needs some insider knowledge (about gflags) before calling it which makes Config really hard to use. Also because it does not what it says it does. The Config class should not be responsible for calling "run". I would propose instead that Config sticks to parsing and storing the config and then main could hand over the Config class to each usecase.

Something like (this is pseudo code but you should get the idea)
```
main(argc, argv) {
    Config cfg(argc, argv);
    switch(cfg.usecase()) {
        case EXEC:
           ExecMonitor exec_monitor;
           exec_monitor.run(cfg);
        default:
          print usage
    }
}
```

In this code, everything sticks to its own role. The config class is responsible for parsing (in its constructor) and storing the config. The caller doesn't need extra knowledge about gflags before using Config, it doesn't need to know much about the different use cases either. Each usecase can just poke at Config when it needs a new config flag, instead of having to store their own copies and having a special constructor with many many arguments etc... 

I'd recommend reading about encapsulation to understand better the role of a class and of an object in C++ https://en.wikipedia.org/wiki/Encapsulation_(computer_programming)

Reply:
One way to think about this is: if someday you need to rewrite the Configuration system, how much do you have to change? With the abstraction that I propose in pseudo-code, you only have to change the content of Config. With your current code, multiple files need to be changed. The class is an abstraction that makes it so that you don't have to think about how it works, you only care about what it does (handle the config).